### ISSUE TITLE
note beam and height 

### ISSUE THREAD

**Issue by YolandaFeng1995** [opensheetmusicdisplay/opensheetmusicdisplay#954]
something wrong with the note beam and height ,
see the screenshot @simonspoerri 
![截屏2021-01-30 下午1 18 41](https://user-images.githubusercontent.com/46066608/106348053-2c564a80-62fe-11eb-85cf-0a6d90244bbb.png)

**Comment by sschmidTU** [comment]
Thanks for the report!
Can you please share a MusicXML sample where this error occurs?
(you can zip it and drag and drop it here)

**Comment by YolandaFeng1995** [comment]
[JTMPC02_51.xml.zip](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/files/5899129/JTMPC02_51.xml.zip)
thx. one more question: the clef change in measure 23 is wrong.

**Comment by sschmidTU** [comment]
Fixed.

![image](https://user-images.githubusercontent.com/33069673/107092563-1e279300-6804-11eb-9013-6ae452379921.png)

The long stems problem was due to wrong stem directions given in the MusicXML file.
It gave different stem directions for notes in the same beam, which in this case is not possible
(unless it were a beam going across staves).
So, the MusicXML export of the program this MusicXML file was created from probably has an error, actually multiple.
Do you know which music program this MusicXML file was created from?

The clef in measure 23 is also not at the wrong position anymore.
Actually we don't render it at all for now (it is not necessary in your sample),
because it is given in a very unfortunate way in the MusicXML that makes it hard to parse,
using `<forward>` and `<backup>` tags, which are impractical and hard to deal with in general.
In the future, we may try to calculate the correct position of clefs when given with these tags.

**Comment by YolandaFeng1995** [comment]
> 固定。
> 
> ![图片](https://user-images.githubusercontent.com/33069673/107093933-7364a400-6806-11eb-8615-90b5d0175dbb.png)
> [osmd_issue_954_beam_stems_mid_measure_clefs_issue_jtmp.musicxml.zip](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/files/5935593/osmd_issue_954_beam_stems_mid_measure_clefs_issue_jtmp.musicxml.zip)
> 
> 长茎问题是由于MusicXML文件中给出的茎方向错误所致。
> 它为同一束中的音符提供了不同的词干方向，在这种情况下这是不可能的，也是错误的
> （只有当它穿过五线谱时才有可能）。
> 因此，创建此MusicXML文件的程序的MusicXML导出可能存在错误，实际上是多个错误。
> 您知道此MusicXML文件是从哪个音乐程序创建的吗？
> 
> 小节23中的谱号也不再位于错误的位置。
> 实际上，我们现在根本不渲染它（在您的示例中是没有必要的），
> 因为它是在MusicXML中非常不幸地给出的，这使得解析，
> 使用`<forward>`和`<backup>`标记变得很困难，这是不切实际的且难以实现处理一般。
> 将来，当给定这些标签时，我们可能会尝试计算谱号的正确位置。

Sibelius
